charts/datastore/templates/deployment.yaml

```diff
@@ -26,13 +26,19 @@ spec:
 {{ toYaml .Values.podAnnotations | indent 8 }}
 {{- end }}
     spec:
+      securityContext:
+        # group ID 1000 is associated with all Containers in the Pod automatically by k8s
+        # we use it to give non-root user access to mounted volumes
+        fsGroup: 1000
```

Comment:
fsGroup should be the same id as the group id you set in the Docker image, and the same group id Kubernetes runs it as (use runAsGroup)